google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/admin/v2/models/CopyBackupRequest.java

```diff
@@ -0,0 +1,144 @@
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import org.threeten.bp.Instant;
+
+/** Fluent wrapper for {@link com.google.bigtable.admin.v2.CopyBackupRequest} */
+public final class CopyBackupRequest {
+  private final com.google.bigtable.admin.v2.CopyBackupRequest.Builder requestBuilder =
+      com.google.bigtable.admin.v2.CopyBackupRequest.newBuilder();
+  private final String sourceBackupId;
+  private final String sourceClusterId;
+  private final String sourceInstanceId;
+  private final String sourceProjectId;
+
+  private String destClusterId;
+
+  /**
+   * Create a {@link CopyBackupRequest} object. It assumes the source backup is located in the same
+   * instance as the destination backup. To copy from a backup in another instance, use {@link
+   * #of(String, String, String) of} method. To copy from a backup in another project, use {@link
+   * #of(String, String, String, String) of} method
+   */
+  public static CopyBackupRequest of(String sourceClusterId, String sourceBackupId) {
+    CopyBackupRequest request = new CopyBackupRequest(sourceClusterId, sourceBackupId, null, null);
+    return request;
+  }
+
+  /**
+   * Create a {@link CopyBackupRequest} object. It assumes the source backup is located in the same
+   * project as the destination backup. To copy from a backup in another project, use {@link
+   * #of(String, String, String, String) of} method
+   */
+  public static CopyBackupRequest of(String sourceClusterId, String sourceBackupId, String sourceInstanceId) {
+    CopyBackupRequest request = new CopyBackupRequest(sourceClusterId, sourceBackupId, sourceInstanceId, null) {
+, null);
+    return request;
+  }
+
+  /**
+   * Create a {@link CopyBackupRequest} object. The source backup could be located in the same or
+   * different cluster and/or instance and/or project.
+   */
+  public static CopyBackupRequest of(
+      String sourceClusterId, String sourceBackupId, String sourceInstanceId, String sourceProjectId) {
+    CopyBackupRequest request = new CopyBackupRequest(sourceClusterId, sourceBackupId, sourceInstanceId, sourceProjectId);
+    return request;
+  }
+
+  private CopyBackupRequest(
+      @Nonnull String sourceClusterId,
+      @Nonnull String sourceBackupId,
+      @Nullable String sourceInstanceId,
+      @Nullable String sourceProjectId) {
+    Preconditions.checkNotNull(sourceBackupId);
+    Preconditions.checkNotNull(sourceClusterId);
+    this.sourceClusterId = sourceClusterId;
+    this.sourceBackupId = sourceBackupId;
+    this.sourceInstanceId = sourceInstanceId;
+    this.sourceProjectId = sourceProjectId;
+  }
+
+  public CopyBackupRequest setDestinationBackupId(String backupId) {
+    Preconditions.checkNotNull(backupId);
+    requestBuilder.setBackupId(backupId);
+    return this;
+  }
+
+  public CopyBackupRequest setDestinationClusterId(String clusterId) {
+    Preconditions.checkNotNull(clusterId);
+    this.destClusterId = clusterId;
+    return this;
+  }
```

Comment:
can dest backupid be set without a dest cluster?
If not, it might be better to have a setDestination(clsuterId, backupId) instead

Reply:
changed this to setDestination(cluster, backup).